rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -100,7 +100,10 @@ def generate_default_string(membset):
                 strlist.append('this->%s.resize(%d);' % (member.name, member.num_prealloc))
             if isinstance(member.default_value, list):
                 if all(v == member.default_value[0] for v in member.default_value):
-                    strlist.append('std::fill(this->%s.begin(), this->%s.end(), %s);' % (member.name, member.name, member.default_value[0]))
+                    # Specifying type for std::fill because of MSBuild warning about cast (C4244)
```

Comment:
I think this comment (and the one below) should probably be more specific; that is, it should specify which version of MSVC we are having problems with, and it should specify that it is only a problem for types that are *smaller* than an int.  That way we don't lose information that we've already discovered :).

Reply:
Ok. Am I correct that we are using MSVC 2017?

Reply:
nvm, I think I got it from the log: **MSVC 14.12**